server/controllers/boards.js

```diff
@@ -0,0 +1,157 @@
+const models = require('../../db/models');
+const dbhelper = require('../../db/helpers.js');
+const helper = require('./helper');
+
+module.exports.getUserOwnedBoards = (req, res) => {
+  if (helper.checkUndefined(req.params.owner_id)) {
+    res.status(400).send('one of parameters from client is undefined');
+  }
+  models.Board.where({ owner_id: req.params.owner_id }).fetch()
+    .then(boards => {
+      if (!boards) {
+        throw 'cant get boards that user owns';
+      }
+      res.status(200).send(boards);
+    })
+    .catch(err => {
+      // This code indicates an outside service (the database) did not respond in time
+      res.status(500).send(err);
+    });
+};
+
+module.exports.getMyBoards = (req, res) => {
+  var userId = req.user.id;
+  dbhelper.getBoardsByUser(userId)
+    .then(boards => {
+      if (!boards) {
+        throw 'user doesnt have any boards';
+      }
+      res.status(200).send(boards);
+    })
+    .catch((err) => {
+      res.status(500).send(err);
+    });
+};
+
+module.exports.createMyBoard = (req, res) => {
+  if (helper.checkUndefined(req.body.board_name, req.body.repo_name, req.body.repo_url)) {
+    res.status(400).send('one of parameters from client is undefined');
+  }
+  var boardObj = {
+    board_name: req.body.board_name,
+    repo_name: req.body.repo_name,
+    repo_url: req.body.repo_url,
+    owner_id: req.user.id
+  };
+  var createdBoard;
+  dbhelper.createBoard(boardObj)
+    .then(result => {
+      if (!result) {
+        throw 'could not create board';
+      }
+      createdBoard = result;
+      return dbhelper.addUserToBoard(req.user.id, result.id);
+    })
+    .then(result => {
+      if (!result) {
+        throw 'could not register user for the board';
+      }
+      res.status(201).send(createdBoard);
+    })
+    .catch(err => {
```

Comment:
createBoard and addUserToBoard already check for duplicates

Reply:
I have removed it.

Reply:
Excellent. It is out of our lives forever